src/ray/raylet/scheduling/policy/affinity_with_bundle_scheduling_policy.cc

```diff
@@ -18,10 +18,33 @@ namespace ray {
 namespace raylet_scheduling_policy {
 
 bool AffinityWithBundleSchedulingPolicy::IsNodeFeasibleAndAvailable(
-    const scheduling::NodeID &node_id, const ResourceRequest &resource_request) {
-  return nodes_.contains(node_id) && is_node_alive_(node_id) &&
-         nodes_.at(node_id).GetLocalView().IsFeasible(resource_request) &&
-         nodes_.at(node_id).GetLocalView().IsAvailable(resource_request);
+    const scheduling::NodeID &node_id,
+    const ResourceRequest &resource_request,
+    bool avoid_gpu_nodes) {
+  if (!(nodes_.contains(node_id) && is_node_alive_(node_id) &&
+        nodes_.at(node_id).GetLocalView().IsFeasible(resource_request) &&
+        nodes_.at(node_id).GetLocalView().IsAvailable(resource_request))) {
+    return false;
+  }
+  if (!avoid_gpu_nodes) {
+    return true;
+  }
+
+  // This function is used by request with no bundle id specified, so we only
+  // focus on the gpu wildcard resource (of the pg) when avoiding gpu nodes.
+  std::string gpu_wildcard_resource_name = "GPU_group_";
```

Comment:
can we create a helper function with unit test for it?

Reply:
I just fixed and simplified this part. I'm not sure if it is worth a dedicated unit test because now it's quite simple. 

BTW, it seems there is not a unit test for `AffinityWithBundleSchedulingPolicy` yet. Let me know if I missed it. @larrylian